client/my-sites/controller.js

```diff
@@ -111,40 +111,57 @@ function renderNoVisibleSites( context ) {
 	);
 }
 
-function isPathAllowedForDomainOnlySite( pathname ) {
-	const urlPrefixesWhiteListForDomainOnlySite = [
-		'/domains/manage/',
-		'/checkout/',
-	];
-
-	return urlPrefixesWhiteListForDomainOnlySite.some( path => startsWith( pathname, path ) );
-}
+function renderSelectedSiteIsDomainOnly( { reactContext, selectedSite } ) {
+	const EmptyContentComponent = require( 'components/empty-content' );
+	const { store: reduxStore } = reactContext;
 
-function onSelectedSiteAvailable( context ) {
-	const selectedSite = sites.getSelectedSite();
-	context.store.dispatch( receiveSite( selectedSite ) );
-	context.store.dispatch( setSelectedSiteId( selectedSite.ID ) );
+	removeSidebar( reactContext );
 
-	const state = context.store.getState();
+	renderWithReduxStore(
+		React.createElement( EmptyContentComponent, {
+			title: i18n.translate( 'This feature is not available for domains' ),
```

Comment:
Alternate string suggestion: 
* `translate( 'This domain is not available.' )`&mdash; translations: **18**. 

[PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/domain-only-site-featuregate)